grid/paste-from-excel/wwwroot/paste-capture.js

```diff
@@ -0,0 +1,72 @@
+﻿var dotNetComponent;
+var containerElem;
+var pasteTextArea;
+
+function initializeExcelPasting(elem, dotnetObject) {
+    if (elem && dotnetObject) {
+        containerElem = elem;
+        dotNetComponent = dotnetObject;
+        // when focus moves to the grid, we will start listening for pasted content
+        // getting the pasted data generaly requires a contenteditable elemnt or an input
+        // using a content editable element is not an option because it would let
+        // people alter and break the components and content inside
+        // so we have to hack an input in only when needed
+        elem.addEventListener("focusin", createTextArea);
+    }
+}
+
+function cleanUpExcelPasting(elem) {
+    if (elem) {
+        elem.removeEventListener();
+        dotNetComponent = null;
+        containerElem = null;
+        destroyTextArea();
+    }
+}
+
+function pasteHandler(e) {
+    setTimeout(function () {
+        if (dotNetComponent) {
+            dotNetComponent.invokeMethodAsync("PasteHandler", pasteTextArea.value);
+            // after pasting, we can remove the textarea to let the user interact with the grid
+            // consider tweaking when and how and where this textarea is depending on your
+            // preferences, UX requirements and needs, and user feedback
+            // here we just clean it up to ensure the same content does not get added more than once
+            pasteTextArea.value = "";
+            //destroyTextArea();
+        }
+    });
+}
+
+function createTextArea() {
```

Comment:
Creating the ghost text area could happen once, when binding the focus handler

Reply:
We can't use clipboardData because that requires an editable HTML element, and that lets the user destroy the grid by changing the html.

The problem with pasting from excel is that you can copy the headers, or you can  avoid copying the headers, so parsing what comes down from the clipboard is a completely heuristic task. I added a sample of that through https://github.com/telerik/blazor-ui/pull/61/commits/f5bd03a8daa3f5b1573c966914c5786da99c2aee

On creating the text area - It could be torn down at any moment due to disposal so I preferred to keep the DOM as clean as possible to also avoid potentially confusing Blazor. Let me know if you want me to refactor this so the textarea only gets removed from the dom on dispose

Reply:
No need to refactor, it's perfect as a starting point to our customers.